### DISCUSSION TITLE
Additional Custom Fields for Issues

### DISCUSSION THREAD

**Discussion by leminhhai** [makeplane/plane#2649] [Ideas]
Currently, our team utilizes various data sources such as RICE, Recommend score (from shareholders), Retention Revenue, and Growth Revenue to determine feature priorities. We have observed that GitLab operates in a similar fashion, but we are looking for a simpler approach. Therefore, we would greatly appreciate the inclusion of custom fields in plane.so that allow us to create additional calculations using formulas (similar to platforms like Baserow, Airtable, LarkSuite).

Having the ability to define custom fields and perform calculations based on these fields would greatly enhance our experience with plane.so. This feature would address the challenge we face in selecting and determining feature priorities, which is often a complex and subjective process. By incorporating custom fields and calculations, we can establish a more structured and data-driven approach to prioritize features accurately.

Furthermore, the flexibility to customize fields and apply calculations would empower us to adapt plane.so to our specific workflow and business requirements. We understand that different organizations have unique sets of metrics and factors when prioritizing features, and the ability to tailor plane.so accordingly would be highly beneficial.

In conclusion, I would like to reiterate the importance of introducing custom fields with calculation capabilities in plane.so. This feature would simplify the process of determining feature priorities and enable us to make more informed decisions. Additionally, the ability to customize fields would ensure plane.so aligns with our specific workflows and allows for seamless integration with our existing processes.

Thank you for considering this additional feedback. We appreciate your commitment to continuously improving plane.so and look forward to the enhancements you will bring to the platform.

**Comment by Huuuze** [comment]
I'd like to add that migrating a project with custom fields from Jira will break without this feature. I recognize the feature request is not a small request, but project flexibility in Plane would be a welcomed addition.

**Comment by wangmengmike** [comment]
Yes, this feature is very required~
To be the real replacement of Jira. We use jira to track business opportunities, adding some mandatory fields to tickets. And we use those fields for analysis and reports (filtering, aggregation, .etc).

**Reply by marcungeschikts** [reply]
I created the issue #2865 related to this discussion which has already a lot of thumbs up.

Feel free to add yours.

**Comment by mmospanenko** [comment]
Will be great to have just plain text fields, without any formulas, as MVP. Formulas are very cool and important, but with just fields we could unblock our workflow (integrate with time tracking systems, for example, add some selectors, labels, etc). And then could be awesome to get formulas like Notion has, for reporting or analytics.

But for now, just custom inputs will be enough 👍

**Comment by vcscroll** [comment]
Will be great to have just plain text fields, without any formulas, as MVP.

**Comment by 17347013442** [comment]
![image](https://github.com/makeplane/plane/assets/52845516/0bbf44dc-a502-45f4-a809-2e5d8e3cba4f)
![image](https://github.com/makeplane/plane/assets/52845516/f7f1fe1e-d56e-49e7-b7ce-9d51ea638b73)
![image](https://github.com/makeplane/plane/assets/52845516/4a7cbdd4-baaa-46bd-924c-4eb22b2048de)
![image](https://github.com/makeplane/plane/assets/52845516/f456e770-9df2-4170-abd8-0152800ed474)
![image](https://github.com/makeplane/plane/assets/52845516/36d8fb1b-c9ff-45bf-96cd-cae56266630a)

**Reply by 17347013442** [reply]
The general feeling is that it can be designed this way

**Comment by MFYDev** [comment]
I do not think I will use plane only if custom field can be added as a feature now.

**Comment by phospholipid77** [comment]
Interestingly, there is a reddit from *one year ago* where a Plane project lead, Vihar, said this: "Custom fields will be shipped under Community Edition soon."
The full text of the comment can be found [in the thread](https://www.reddit.com/r/selfhosted/comments/13z5xr4/comment/jmqs6a5/).

**Comment by rafaellehmkuhl** [comment]
Any news on this?

I totally agree with others about the calculations being a stretch. The main feature is definitely the ability to add a custom text field. This probably solves 80% of people's needs.

I'm using it for my personal activities and this is truly the only feature I'm missing from competition.
If you accept contributions, I could happily help.

**Comment by marcungeschikts** [comment]
Maybe I do not understand it correctly 🤷🏻, but it seems that custom fields (custom issues + properties) will only be available with the Plane Pro license.  😢 
![image](https://github.com/user-attachments/assets/2ea268ee-379d-4d1f-8163-8aba083ad49c)

**Reply by phospholipid77** [reply]
Is a custom "issue" different from a custom "field"? I think a custom field is different. For me a "custom field" is a single variable in a task or issue.  

If for them a custom "issue" translates to a custom "field", then I'm not sure what this comment on Reddit was intended to communicate. This comment seems to make clear an intention to release custom fields as a feature to the community. 

I mean if they don't, they don't. I have utmost respect for these types of project. I also don't have a lot of patience to grow alongside of them. So, I'll check back in a year after my ClickUp subscription is done. 

![Screen Shot 2024-07-30 at 12 22 57 PM](https://github.com/user-attachments/assets/bf5fce4e-279a-4d2c-9d64-2fe320a31512)

**Reply by UbiquitousBear** [reply]
@vihar can you provide any update and context to the above? Also, the value of Plane One is become less clear with the addition of the Pro tier.

**Comment by mmospanenko** [comment]
Plane is flying away...

**Comment by StreamlinedStartup** [comment]
+100

**Comment by henriquelatorre** [comment]
I believe the existing issue types and custom fields should be core features available in the community release. I understand that elevating some basic features to the pro level can be necessary while other pro-only stuff is still baking.

As a suggestion for the commercial offering, I recommend restricting to subscription-only the features that leverage custom fields. These could include using custom fields for conveniences such as automations and other advanced manipulations.

**Reply by vcscroll** [reply]
It's core feature for the product, please make plane community driven again!

**Comment by vcscroll** [comment]
It's core feature for the product, please make plane community driven again!
